routes/movies.js

```diff
@@ -43,83 +37,114 @@ router.post('/', async (request, response, next) => {
+      console.error('Error receiving movies from database')
+      return next({
+        status: 404,
+        message: 'Error receiving movies from database',
+      })
+    }
+    response.status(200).json(movies)
+  } catch (error) {
+    console.error(error)
+    return next({ status: 500, message: 'Server error' })
+  }
 })
 
-router.get('/top', (request, response, next) => {
-  Movie.find()
-    .then(data => {
-      /*
-        sort data by rating
-        return the data with max 100 entries
-      */
-      data.sort((a, b) => b.rating - a.rating).slice(0, 100)
-
-      response.status(200).json(data)
-    })
-    .catch(error =>
-      next({ status: 404, message: error.message || 'No documents found' })
-    )
+router.get('/top', async (request, response, next) => {
+  try {
+    const movies = await Movie.find()
+    if (!movies) {
+      console.error('Error receiving movies from database')
+      return next({
+        status: 404,
+        message: 'Error receiving movies from database',
+      })
+    }
+    movies.sort((a, b) => b.rating - a.rating).slice(0, 100)
+    response.status(200).json(movies)
+  } catch (error) {
+    console.error(error)
+    return next({ status: 500, message: 'Server error' })
+  }
 })
 
-router.get('/voting', (request, response, next) => {
-  Movie.find()
-    .then(data => {
-      /*
-        get an random index from the data array
-        add it to firstMovieIndex
-        do this for the second movie
-        repeat until firstMovieIndex and secondMovieIndex arent equal
-        return new array with both movies
-      */
-      if (data.length < 2) {
-        const error = { message: 'Not enough movies in the database.' }
-        return next({ status: 404, message: error.message })
-      }
+router.get('/voting/:id', async (request, response, next) => {
+  const { id } = request.params
 
-      const firstMovieIndex = Math.floor(Math.random() * data.length)
-      let secondMovieIndex = firstMovieIndex
-      let gotSecondMovie = false
+  try {
+    const user = await User.findById(id)
+    if (!user) {
+      console.error('User does not exist')
+      return next({ status: 404, message: 'User does not exist' })
+    }
 
-      do {
-        secondMovieIndex = Math.floor(Math.random() * data.length)
-        if (firstMovieIndex !== secondMovieIndex) {
-          gotSecondMovie = true
-        }
-      } while (!gotSecondMovie)
+    const movies = await Movie.find()
+    if (!movies) {
+      console.error('Error receiving movies from database')
+      return next({
+        status: 404,
+        message: 'Error receiving movies from database',
+      })
+    }
 
-      const dataVoting = [data[firstMovieIndex], data[secondMovieIndex]]
+    if (movies.length - user.unwatchedMovies.length < 2) {
+      const error = { message: 'Not enough movies in the database.' }
+      console.error(error.message)
+      next({ status: 400, message: error.message || 'No documents found' })
```

Comment:
```suggestion
      next({ status: 409, message: error.message || 'No documents found' })
```
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409